tasks/libs/junit_upload.py

```diff
@@ -246,3 +250,32 @@ def repack_macos_junit_tar(infile, outfile):
         # pack all files to a new tarball
         for f in os.listdir(tempd):
             outfp.add(os.path.join(tempd, f), arcname=f)
+
+
+def retrieve_jira_card(xmlfile_path):
+    """
+    Search in jira if a card already exist for the given testsuite
+    """
+    from .pipeline_notifications import send_slack_message
+    from jira import JIRA
+
+    tree = ET.parse(xmlfile_path)
+    auth = ("mail", "token")
+    j = JIRA(basic_auth=auth, server="https://datadoghq.atlassian.net/")
+    test_suites = list(tree.iter("testsuite"))
+    if len(test_suites) > 1:
+        message = f"Found several test suites in a single junit file: {[x.attrib['name'] for x in test_suites]}"
+        send_slack_message("#agent-test-alerts", message)
+    suite_name = test_suites[0].attrib["name"]
+    search_query = f'project = "CI Improvement" and summary ~ "{suite_name}" and status != Done'
+    try:
+        issues = j.search_issues(search_query)
```

Comment:
I should maybe add a retry here

Reply:
If we fail finding the card we must not fail the job, so I put a try/except to send a fix value that is managed during the [workflow execution](https://app.datadoghq.com/workflow/42375aaf-9a77-4b93-ad51-9a5f524b570d)

Reply:
@spencergilbert just checked and the JIRA API already has a [3-retries policy](https://github.com/pycontribs/jira/blob/3.5.2/jira/client.py#L400) with 60s sleep time